### ISSUE TITLE
Accessibility: button fails  colour contrast - WCAG 1.4.3 

### ISSUE THREAD

**Issue by chris-gds** [chakra-ui/chakra-ui#8019]
### Description

When I visit the button page
https://chakra-ui.com/docs/components/button
I expect the colours to pass contrast checks for Accessibility however there are issues.

## Screenshot of buttons

![chakra-buttons screenshot of UI](https://github.com/chakra-ui/chakra-ui/assets/71266765/d4861e5b-1d5f-48d2-acfd-cbde02a2b7c5)

---

## Failing colour contrast checks

![chakra-colour-contrast checks, fails WCAG 2.1 AA](https://github.com/chakra-ui/chakra-ui/assets/71266765/39d88d4f-3b29-49f4-b98d-c79836ca16b4)

### Link to Reproduction

https://chakra-ui.com/docs/components/button

### Steps to reproduce

Go to: https://chakra-ui.com/docs/components/button

### Chakra UI Version

Latest

### Browser

All

### Operating System

- [X] macOS
- [X] Windows
- [X] Linux

### Additional Information

It doesn't just impact the default buttons, many of these other colour examples fail WCAG 1.4.3 too.

![screenshot of other buttons on chakra](https://github.com/chakra-ui/chakra-ui/assets/71266765/40cf53f8-830b-48f6-a604-b018abb7e965)

--- 

Consider a check on Linkedin:
![LinkedinCheck](https://github.com/chakra-ui/chakra-ui/assets/71266765/58b6428f-a40c-4068-ba85-285d41a10dbb)

## WCAG documentation 

https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html

**Comment by segunadebayo** [comment]
This is a design flaw in Chakra v2. We've improved this significantly in v3.
